### ISSUE TITLE
[BUG] unable to convert the HuggingFace model to the ServerlessLLM format based on the documentation

### ISSUE THREAD

**Issue by bwj1993** [ServerlessLLM/ServerlessLLM#90]
### Prerequisites

- [X] I have read the [ServerlessLLM documentation](https://serverlessllm.github.io/).
- [X] I have searched the [Issue Tracker](https://github.com/ServerlessLLM/ServerlessLLM/issues) to ensure this hasn't been reported before.

### System Information

Based on the base image (nvcr.io/nvidia/pytorch:24.08-py3), Conda was installed, as well as the sllm-store mentioned in the documentation.

### Problem Description

According to the “Usage with vLLM” section in the https://serverlessllm.github.io/docs/stable/store/quickstart#usage-with-vllm document, when attempting to load a local HuggingFace model and convert it into the ServerlessLLM format, the following error occurred:
<img width="1503" alt="e745f1c60aaf0d2aded64b5b8" src="https://github.com/user-attachments/assets/9381f58e-b5f3-4036-b72b-59af56cf28bc">

It seems that the patch is effective in the source code but not in the Conda package. Upon checking the /root/miniconda3/envs/sllm-store/lib/python3.10/site-packages/serverless_llm_store/init.py file, indeed there is no information related to save_dict.

How should I proceed to ensure that the patch is applied to the /root/miniconda3/envs/sllm-store/lib/python3.10/site-packages/serverless_llm_store/init.py file?

### Steps to Reproduce

Code snippets:
```python
import os
import shutil
from typing import Optional

class VllmModelDownloader:
    def __init__(self):
        pass

    def download_vllm_model(
        self,
        model_name: str,
        torch_dtype: str,
        tensor_parallel_size: int = 1,
        pattern: Optional[str] = None,
        max_size: Optional[int] = None,
    ):
        import gc
        import shutil
        from tempfile import TemporaryDirectory

        import torch
        from huggingface_hub import snapshot_download
        from vllm import LLM
        from vllm.config import LoadFormat

        def _run_writer(input_dir, output_dir):
            # load models from the input directory
            llm_writer = LLM(
                model=input_dir,
                download_dir=input_dir,
                dtype=torch_dtype,
                tensor_parallel_size=tensor_parallel_size,
                num_gpu_blocks_override=1,
                enforce_eager=True,
                max_model_len=1,
            )
            model_executer = llm_writer.llm_engine.model_executor
            # save the models in the ServerlessLLM format
            model_executer.save_serverless_llm_state(
                path=output_dir, pattern=pattern, max_size=max_size
            )
            for file in os.listdir(input_dir):
                # Copy the metadata files into the output directory
                if os.path.splitext(file)[1] not in (
                    ".bin",
                    ".pt",
                    ".safetensors",
                ):
                    src_path = os.path.join(input_dir, file)
                    dest_path = os.path.join(output_dir, file)
                    if os.path.isdir(src_path):
                        shutil.copytree(src_path, dest_path)
                    else:
                        shutil.copy(src_path, output_dir)
            del model_executer
            del llm_writer
            gc.collect()
            if torch.cuda.is_available():
                torch.cuda.empty_cache()
                torch.cuda.synchronize()

        # set the model storage path
        storage_path = os.getenv("STORAGE_PATH", "./models/vllm")
        model_dir = os.path.join(storage_path, model_name)

        # create the output directory
        if os.path.exists(model_dir):
            print(f"Already exists: {model_dir}")
            return
        os.makedirs(model_dir, exist_ok=True)

        try:
            with TemporaryDirectory() as cache_dir:
                # download model from huggingface
                #input_dir = snapshot_download(
                #    model_name,
                #    cache_dir=cache_dir,
                #    allow_patterns=["*.safetensors", "*.bin", "*.json", "*.txt"],
                #)
                _run_writer("/mnt/pfs/huggingface/facebook/opt-1.3b", model_dir)
        except Exception as e:
            print(f"An error occurred while saving the model: {e}")
            # remove the output dir
            shutil.rmtree(model_dir)
            raise RuntimeError(
                f"Failed to save model {model_name} for vllm backend: {e}"
            )

downloader = VllmModelDownloader()
downloader.download_vllm_model("facebook/opt-1.3b", "float16", 1)
```

Steps to reproduce:
1. docker  pull nvcr.io/nvidia/pytorch:24.08-py3
2. docker run --gpus all -v /mnt/nvme/:/mnt/nvme/-v /mnt/nfs:/mnt/nfs  -td nvcr.io/nvidia/pytorch:24.08-py3  bash
3. docker exec -it xxx bash
4. Refer to https://docs.anaconda.com/miniconda/ to install Conda.
5. conda create -n sllm-store python=3.10 -y
6. conda activate sllm-store
7. pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ serverless_llm_store==0.0.1.dev3
8. cd ServerlessLLM/serverless_llm/store
9. git pull --rebase
10. ./serverless_llm/store/vllm_patch/patch.sh
11. python convert_model_facebook_vllm.py

### Expected Behavior

Successfully loaded a local HuggingFace model  and save it in the ServerlessLLM format

**Comment by SiyangShao** [comment]
Hi, I think the reason is that the released `serverless_llm_store` is too old, and in that version, integration with vLLM hasn't been done. The patch file is only for vLLM's installed lib, and it will not change the serverlessLLM.
Could you try installing the `serverless_llm_store` from [source](https://serverlessllm.github.io/docs/stable/store/quickstart#install-from-source)? We'll release the pre-built wheel for our latest version soon. @future-xy

**Comment by bwj1993** [comment]
I performed a patch operation and passed the check using the ./serverless_llm/store/vllm_patch/check_patch.sh command. Then, I executed the pip install . command. However, upon rechecking /root/miniconda3/envs/sllm-store/lib/python3.10/site-packages/serverless_llm_store/__init__.py, save_dict is still missing.

<img width="1503" alt="38dfcacb572a5c77c2edfd4ac" src="https://github.com/user-attachments/assets/37d7bea1-0ace-4195-840f-23fe06eb9786">

<img width="1387" alt="40381c060eda9b681d023ae2d" src="https://github.com/user-attachments/assets/9ab78b3f-5776-4a44-ab70-c67fa610d42b">

**Comment by SiyangShao** [comment]
Hi, I noticed that it seems that you are running `pip install .` under our project folder and installing the `serverless_llm` from the source. But actually, the checkpoint loader (aka `serverless_llm_store`) is an independent component of serverlessLLM, and it's under the `serverless_llm/store` folder. Could you try to install the `serverless_llm_store` part? 
If you only install `serverless_llm` from the source, it will still download the `serverless_llm_store` from our prebuilt wheel, which is too old.

**Comment by SiyangShao** [comment]
Here's the correct info under `__init__.py`:
```
from .sllm_store import load_dict, load_model, save_model, save_dict, load_dict_single_device

__all__ = ["load_model", "save_model", "load_dict", "save_dict", "load_dict_single_device"]
```

**Comment by bwj1993** [comment]
However, when I execute the pip install . command in the serverless_llm/store directory, the following error is reported:
<img width="1503" alt="f32d10260abd4d84fae0a93c1" src="https://github.com/user-attachments/assets/76c3dfed-0ce0-46e4-a7a2-319efa1ebf65">
Do I need to execute pip install on a machine with a GPU?

**Comment by SiyangShao** [comment]
I think this error is usually because the last build cache hasn't been cleaned successfully, so the pip was directed into the wrong dir.
`pip install .` on a machine without GPU (but with CUDA) should be ok, but if you want to test it, GPU is necessary.

**Comment by bwj1993** [comment]
Unfortunately, our machines with internet access do not have CUDA and GPUs, and those with CUDA and GPUs cannot access the internet. Are there any other ways to install serverless_llm/store with vllm?

**Comment by future-xy** [comment]
> Unfortunately, our machines with internet access do not have CUDA and GPUs, and those with CUDA and GPUs cannot access the internet. Are there any other ways to install serverless_llm/store with vllm?

Hi! Are you able to install `serverless_llm_store` via `pip` on your GPU machines? Currently, to convert a checkpoint into vLLM supported format, GPUs are necessary. 
If you can install via `pip`, we will update the testpypi repo soon to reflect recent changes. However, in this case, you might still need to copy and apply [the patch file](https://github.com/ServerlessLLM/ServerlessLLM/blob/main/serverless_llm/store/vllm_patch/sllm_load.patch) on your GPU machines where `serverless_llm_store` is installed.

**Comment by future-xy** [comment]
> Currently, I install serverless_llm_store via pip on a CPU machine with internet access, then build it into an image. I launch the built image on a GPU machine for testing. When is the testpypi repository expected to be updated?

Thank you for your update! Yes, you should be able to convert the checkpoints after launching the built image on your GPU machine. The TestPyPI repository is expected to be updated within the next couple of hours.

Let me know if you have any other questions!

**Comment by bwj1993** [comment]
Okay, thank you very much. If the TestPyPI repository is updated, please also update the documentation accordingly. I will follow the steps in the documentation to rebuild an image and then conduct tests.

**Comment by future-xy** [comment]
> Okay, thank you very much. If the TestPyPI repository is updated, please also update the documentation accordingly. I will follow the steps in the documentation to rebuild an image and then conduct tests.

Hi! We've just released and tested `serverless_llm_store==0.0.1.dev4` ([TestPyPI link](https://test.pypi.org/project/serverless-llm-store/0.0.1.dev4/)). This version should address the initial issue you encountered. Please don't hesitate to reach out if you run into any further issues. Thank you for your valuable feedback!

**Comment by 5symx** [comment]
Hi there, Similar issue for me. Solved by install another version.

Current store version `serverless_llm_store=0.0.1.dev5` from the [Document](https://serverlessllm.github.io/docs/stable/store/quickstart/) occurs the following error, when i running the template on the quick start guide.
```
Traceback (most recent call last):
  File "FILENAME.py", line 1, in <module>
    from serverless_llm_store import save_model
ImportError: cannot import name 'save_model' from 'serverless_llm_store' (FILEPATH/miniconda3/envs/sllm-store/lib/python3.10/site-packages/serverless_llm_store/__init__.py)
```
After I downgrade to `serverless_llm_store=0.0.1.dev4`. It works fine for me.

**Comment by future-xy** [comment]
> Hi there, Similar issue for me. Solved by install another version.
> 
> Current store version `serverless_llm_store=0.0.1.dev5` from the [Document](https://serverlessllm.github.io/docs/stable/store/quickstart/) occurs the following error, when i running the template on the quick start guide.
> 
> ```
> Traceback (most recent call last):
>   File "FILENAME.py", line 1, in <module>
>     from serverless_llm_store import save_model
> ImportError: cannot import name 'save_model' from 'serverless_llm_store' (FILEPATH/miniconda3/envs/sllm-store/lib/python3.10/site-packages/serverless_llm_store/__init__.py)
> ```
> 
> After I downgrade to `serverless_llm_store=0.0.1.dev4`. It works fine for me.

Thanks for your feedback! The issue you're encountering stems from a compatibility problem between the current code and the documentation. We’ve addressed this, and a fix has already been submitted in PR #104.
